MapboxCoreNavigation/RouteController.swift

```diff
@@ -530,21 +530,21 @@ extension RouteController: CLLocationManagerDelegate {
     }
 
     @objc public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        guard let location = locations.last else {
-            return
-        }
-        self.rawLocation = location
-
-        sessionState.pastLocations.push(location)
-
-        if location.isQualified {
+        let filteredLocations = locations.filter { $0.isQualified }
+        
+        if !filteredLocations.isEmpty, hasFoundOneQualifiedLocation == false {
```

Comment:
Nit: `== false` is unnecessarily explicit in Swift. The `!` operator would suffice.